site/en/external/faq.md

```diff
@@ -12,6 +12,71 @@ Bazel.
 
 ## MODULE.bazel {:#module-bazel}
 
+### How should I version a Bazel module? {:#module-versioning-best-practices}
+
+Setting `version` with the [`module`](/rules/lib/globals/module#module)
+directive in the source archive `MODULE.bazel` can have several downsides and
+unintended side effects if not managed carefully:
+
+*   Duplication: releasing a new version of a module typically involves both
+    incrementing the version in `MODULE.bazel` and tagging the release, two
+    separate steps that can easily fall out of sync. While automation can reduce
+    this risk, it's simpler and safer to avoid it altogether.
+
+*   Inconsistency: users overriding a module with a specific commit via a
+    [non-registry override](module.md#non-registry_overrides) will see an
+    incorrect version. For instance, if the `MODULE.bazel` in the source
+    archive sets `version = "0.3.0"` but additional commits have been made
+    since that release, a user overriding with one of those commits would still
+    see `0.3.0`. In reality, the version should reflect that it's ahead of the
+    release, e.g. `0.3.1-rc1`.
+
+*   Non-registry override issues: using placeholder values can cause issues when
+    users override a module with a non-registry override. For example, `0.0.0`
+    doesn't sort as the highest version, which is usually the expected behavior
+    users want when doing a non-registry override.
+
+Thus, it's best to avoid setting the version in the source archive
+`MODULE.bazel`. Instead, set it in the `MODULE.bazel` stored in the registry
+(e.g., the [Bazel Central Registry](https://registry.bazel.build/)), which is
+the actual source of truth for the module version during Bazel's external
+dependency resolution (see [Bazel
+registries](https://bazel.build/external/registry)).
+
+This is usually automated, e.g. the
+[`rules-template`](https://github.com/bazel-contrib/rules-template) example rule
+repository uses a [bazel-contrib/publish-to-bcr publish.yaml Github
+Action](https://github.com/bazel-contrib/publish-to-bcr/blob/e5482809f9b5d26fb22c8341055d4f2ddd1dec46/.github/workflows/publish.yaml)
+to publish the release to the BCR.
+
+The action generates a patch for the source archive `MODULE.bazel` with the
+release version. This patch is stored in the registry and is applied when the
+module is fetched during Bazel's external dependency resolution.
```

Comment:
Just FWIW, the relevant `publish-to-bcr` code (in case you want to link it or anything):

- https://github.com/bazel-contrib/publish-to-bcr/blob/v0.2.2/src/domain/create-entry.ts#L176-L216
- https://github.com/bazel-contrib/publish-to-bcr/blob/v0.2.2/src/domain/module-file.ts#L35-L49
- https://github.com/bazel-contrib/publish-to-bcr/blob/v0.2.2/src/domain/module-file.spec.ts#L63-L164